src/braket/emulation/base_emulator.py

```diff
@@ -0,0 +1,79 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from __future__ import annotations
+
+from collections.abc import Iterable
+from typing import Optional, Union
+
+from braket.passes import BasePass, ProgramType, ValidationPass
+
+
+class BaseEmulator:
```

Comment:
Thinking about it more, this class should be renamed; it's not doing any emulation at all, but transforming or validating the payload. Maybe `ProgramTransformer` or `PassManager`?